server/auth/auth.go

```diff
@@ -434,52 +517,59 @@ func (auth *AuthManager) ExpectUsers(users map[account.AccountID]struct{}, withi
 // RecordCancel records a user's executed cancel order, including the canceled
 // order ID, and the time when the cancel was executed.
 func (auth *AuthManager) RecordCancel(user account.AccountID, oid, target order.OrderID, epochGap int32, t time.Time) {
-	auth.recordOrderDone(user, oid, &target, epochGap, t.UnixMilli())
+	score := auth.recordOrderDone(user, oid, &target, epochGap, t.UnixMilli())
+
+	tier, bondTotal := auth.computeUserTier(user, score)
+	log.Debugf("User %v ban score %d, bond total %v => tier %v", user, score, bondTotal, tier)
+	// If their tier sinks below 1, unbook their orders and send a note.
+	if tier < 1 {
+		details := fmt.Sprintf("excessive cancellation rate, new tier = %d", tier)
+		auth.Penalize(user, account.CancellationRate, details)
+	}
 }
 
 // RecordCompletedOrder records a user's completed order, where completed means
 // a swap involving the order was successfully completed and the order is no
 // longer on the books if it ever was.
 func (auth *AuthManager) RecordCompletedOrder(user account.AccountID, oid order.OrderID, t time.Time) {
-	auth.recordOrderDone(user, oid, nil, db.EpochGapNA, t.UnixMilli())
+	score := auth.recordOrderDone(user, oid, nil, db.EpochGapNA, t.UnixMilli())
+	auth.computeUserTier(user, score) // may raise tier, but no action needed
```

Comment:
Ah, yes!  I defined a `TierChangedRoute`, but marked it's use as a TODO
https://github.com/decred/dcrdex/blob/33986669e42f7ddc35d53b2eb80d27f3ee6aebd0/dex/msgjson/types.go#L177-L179

Might as well wire that up in this work.

Reply:
I think this might be the last pending change I have for this PR.  Will wire up a `tierchange` handler on client that notifies, and have server send the ntfn spots like the one on which we're commenting presently.

Reply:
Still planning to wire up the `tierchange` notes